src/Sulu/Bundle/ContentBundle/Serializer/Subscriber/StructureSubscriber.php

```diff
@@ -113,16 +113,13 @@ private function addStructureProperties(StructureMetadata $structureMetadata, St
     {
         /** @var ManagedStructure $structure */
         $structure = $document->getStructure();
-        foreach ($structure->toArray() as $name => $value) {
-            if ($name === 'title') {
+        $data = $structure->toArray();
+        foreach ($structureMetadata->getProperties() as $name => $property) {
+            if ($name === 'title' || !array_key_exists($name, $data) || $property->hasTag('sulu.rlp')) {
```

Comment:
because in some reasons you get not a managed Structure (which only returns values for existing properties). you get a structure which also returns for example shadowOn as a structure property. which is actually not but the mapping create a structure property-value for it. to be sure that such values are not published over the api this logic was introduced.